samza-core/src/test/java/org/apache/samza/clustermanager/MockContainerAllocator.java

```diff
@@ -23,16 +23,29 @@
 import java.lang.reflect.Field;
 
 import java.util.Map;
+import java.util.concurrent.CountDownLatch;
 
 public class MockContainerAllocator extends ContainerAllocator {
   public int requestedContainers = 0;
+  private CountDownLatch latch = new CountDownLatch(1);
 
   public MockContainerAllocator(ClusterResourceManager manager,
                                 Config config,
                                 SamzaApplicationState state) {
     super(manager, config, state);
   }
 
+  /**
+   * Causes the current thread to block until the provided number of containers have started.
+   *
+   * @param numExpectedContainers  the number of containers expected to start
+   * @throws InterruptedException  if the current thread is interrupted while waiting
+   */
+  void awaitContainersStart(int numExpectedContainers) throws InterruptedException {
+    latch = new CountDownLatch(numExpectedContainers);
```

Comment:
1. `awaitContainersStart` is not meant to be used in a multi-threaded context. 

2. The only guarantee is that, we will wait until 2 containers have started after the call to `awaitContainersStart`. The caller should ensure that `awaitContainerStart` is called before any action that triggers the startup. This is just an alternative to doing a `Thread.sleep()`